detect.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/paketo-buildpacks/packit"
+)
+
+type BuildPlanMetadata struct {
+	Launch bool `toml:"launch"`
+}
+
+//go:generate faux --interface Parser --output fakes/parser.go
+type Parser interface {
+	Parse(path string) (hasMri bool, err error)
+}
+
+func Detect(gemfileParser Parser) packit.DetectFunc {
+	return func(context packit.DetectContext) (packit.DetectResult, error) {
+		_, err := os.Stat(filepath.Join(context.WorkingDir, "config.ru"))
+		if err != nil {
+			if os.IsNotExist(err) {
+				return packit.DetectResult{}, packit.Fail
+			}
+			return packit.DetectResult{}, fmt.Errorf("failed to stat config.ru: %w", err)
+		}
+
+		hasMri, err := gemfileParser.Parse(filepath.Join(context.WorkingDir, "Gemfile"))
+		if err != nil {
+			return packit.DetectResult{}, fmt.Errorf("failed to parse Gemfile: %w", err)
+		}
+
+		if !hasMri {
+			return packit.DetectResult{}, packit.Fail
+		}
```

Comment:
I don't think this should fail if the app isn't using MRI. We are going to be adding `jruby` as a possible Ruby implementation at some point in the near future and this would break.

Reply:
Per current design in this PR - this cnb requires `mri` and this was done to be consistent with that requirement. What's an alternate way for the cnb to say that it requires some ruby interpreter? Are we going to be using the same `provides` for all ruby interpreters?

Reply:
I guess this is fine for now. I think it feels more natural to just require `mri` and then have the buildpack group fail if that doesn't work. I think this same rule would apply to the `puma` buildpack. I'll make a PR to that to show what make sense to me.

Reply:
I've added a PR for `puma`, https://github.com/paketo-community/puma/pull/8, and explained a bit more about why I think this check is not required for now.